ruby/basics/loops.md

```diff
@@ -17,35 +20,111 @@ what the student is expected to know or be able to do by the end of this lesson
 * You know how to use the `upto` and `downto` loops in ruby
 
 ### Loop
-* Rubys loop loop lol
-* what its good for
-* how to break out of it
-* what its good for
+The loop loop (say what????) is Ruby's loop that just don't quit. It's an infinite loop that'll keep going unless you specifically request for it to stop using the `break` command. Break is commanly used with a condition as illustrated in the code example below.
+
+```ruby
+i = 0
+loop do
+  puts "i is #{i}"
+  i += 1
+  break if i == 10
+end
+```
+
+You won't see this loop used much in Ruby and if you're using it then there is probably a better loop for you out there.
 
 ### While Loop
-* how it works
-* include code snippets for a while loop that loops a certain amount of times and one that loops until a condition is met.
+
+A while loop is similar to the `loop` loop except you declare the condition that will break out of the loop up front.
+
+```ruby
+i = 0
+while(i < 10) do
+ puts "i is #{i}"
+ i += 1
+end
+```
+This is an example of using a while loop with a count, and as you declare the condition that breaks the loop up-front, the intention of your code is much clearer making it easier to read than our loop loop above.
+
+While loops can also be used to repeatedly ask a question of the user until they give an appropriate response.
+
+```ruby
+while gets.chomp != "yes" do
+  puts "Will you go to prom with me?"
+end
+```
+In real life you obviously should just take no for an answer the first time.
+
 
 ### For Loop
-* how it works
-* a few code snippets
-* mention its barely used in favour of enumerables
+
+For loops are used to iterate through a collection of information such as an array or range.
+
+```ruby
+for i in 0..5
+  puts "The number of zombies I'd take out before succumbing is #{i}"
+end
+```
+It's really all there is to it.
 
 ### Times Loop
-* how it works
-* code snippets
-* what its good for
+If you need to run a loop a specified number of times then look no further than the trusty times loop. It works by iterating through a loop a number of times and even throws in the bonus of accessing the number it's currently iterating through.
+
+```ruby
+5.times do
+  puts "Hello, world!"
+end
+```
+I'm sure you can guess what that code does. Ruby's readable that way!
+
+```ruby
+5.times do |number|
+  puts "Alternative fact number #{number}"
+end
+```
+Remember, loops will start counting from a zero index unless specified otherwise so the first loop iteration will output `Alternative fact number 0`.
```

Comment:
Might be helpful to type out the iterations, specifically to show that the last element will NOT be 5.